src/Utils/file_utils.py

```diff
@@ -0,0 +1,60 @@
+import os
+import glob
+import pandas as pd
+
+from langchain.text_splitter import RecursiveCharacterTextSplitter, Language
+
+
+def read_source_code_file(path):
+    with open(path, "r", encoding='utf-8') as f:
+        text_splitter = RecursiveCharacterTextSplitter.from_language(language=Language.C,
+                                                                     chunk_size=100, chunk_overlap=0)
+        plain_text = f.read()
+        doc_list = text_splitter.create_documents([plain_text])
+        return doc_list
+
+def create_embeddings_for_all_project_files():
+    res_list = []
+    # reading project src folder
+    src_dir_path = os.path.join(os.getcwd(), "systemd-rhel9/src/")
+    count = 0
+    for src_filename in glob.iglob(src_dir_path + '/**/**', recursive=True):
+
+        if (src_filename.endswith(".c") or src_filename.endswith(".h")) and os.path.isfile(src_filename):
+            count = count + 1
+            for k in read_source_code_file(src_filename):
+                res_list.append(k.page_content)  # adding source code file as text to embeddings
+    print(f"Total files: {count}")
+    return res_list
+
+def read_known_errors_file(path):
+    with open(path, "r", encoding='utf-8') as f:
+        text_splitter = RecursiveCharacterTextSplitter(separators=["\n\n", "\n"],
+                                                       chunk_size=500, chunk_overlap=0)
+        plain_text = f.read()
+        doc_list = text_splitter.create_documents([plain_text])
+        return doc_list
+    
+def get_human_verified_results():
+    filename = os.getenv("HUMAN_VERIFIED_FILE_PATH")  
+    try:
+        df = pd.read_excel(filename)
+    except Exception as e:
+        raise ValueError(f"Failed to read Excel file at {filename}: {e}")
+    
+    df.columns = df.columns.str.strip().str.lower()
```

Comment:
In the case of file read fails, does this statement works fine? Because it is completely fine to run the process without a human verified results.